app/models/zone.rb

```diff
@@ -48,7 +55,29 @@ def find_master_server
     active_miq_servers.detect(&:is_master?)
   end
 
+  def self.create_maintenance_zone
+    begin
+      # 1) Create Maintenance zone
+      zone = new(:name        => "#{MAINTENANCE_ZONE_NAME_PREFIX}#{SecureRandom.uuid}",
+                 :description => "Maintenance Zone",
+                 :visible     => false)
+
+      zone.save!
+
+      # 2) Assign to MiqRegion
+      MiqRegion.seed if MiqRegion.my_region.nil?
```

Comment:
I don't think we could get here without a region.  If you need to create one for test purposes, please do that in the spec.

Reply:
Without it you'll depend on order of seeding - on some not straightforward code. Why to create unrelated dependencies if not necessary? 
It's the same like one method believes that parameter is not nil - which is obviously wrong too.

Reply:
This is less of an issue I think now that we're only calling this from `Zone.seed` instead of from the cache_with_timeout method, however since what we're doing here is ensuring the order of seeds being run it would be more obvious if we raised if MiqRegion.my_region is nil, that way it is more obvious if someone changes the order of the seeding.

Reply:
You don't need the check at all... MiqRegion.seed is idempotent

Reply:
I'm not sure what's conclusion from this discussion. 
I remove creating from `cache_with_timeout` and then calls MiqRegion.seed there without checking